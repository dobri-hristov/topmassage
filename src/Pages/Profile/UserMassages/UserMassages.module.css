.title {
    text-align: center;
}

.tr {
    display: flex;
    justify-content: space-between;
    padding: 0px 15px;
}

.trMain h3,
.title {
    margin-top: 5px;
    margin-bottom: 10px;
    color: #673AB7;
}

.type {
    flex-basis: 60%;
}

.massages p {
    margin-top: 4px;
    margin-bottom: 4px;
}

.date {
    text-align: center;
}

.hour {
    margin-left: 10px;
}

.more {
    margin: 10px 0 5px 0;
    color: #673AB7
}

@media screen and (max-width: 390px) {
    .trMain {
        display: none;
    }

    .tr {
        flex-direction: column;
        text-align: center;
    }
}