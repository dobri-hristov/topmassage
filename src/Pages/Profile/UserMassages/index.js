import React from 'react'
import styles from './UserMassages.module.css'
import sortDate from './functions/sortDate'
import filterDate from './functions/filterDate'

const UserMassages = ({ massages }) => {
    if (massages.length !== 0) {
        massages = massages.filter(massage => filterDate(massage.date))
    }

    return (
        <div className="massages-form">
            <h3 className={styles.title}>Предстоящи масажи</h3>
            <div className={`${styles.tr} ${styles.trMain}`}>
                <h3 className={styles.type}>Вид масаж</h3>
                <h3>Дата</h3>
                <h3 className={styles.hour}>Час</h3>
            </div>
            {
                massages.length !== 0
                    ? massages
                        .sort((a, b) => sortDate(a, b))
                        .slice(0, 5)
                        .map((res, index) => (
                            <div className={`${styles.tr} ${styles.massages}`} key={index}>
                                <p className={styles.type}>{index + 1}. {res.massage}</p>
                                <p className={styles.date}>{res.date}</p>
                                <p className={styles.hour}>{res.hour}</p>
                            </div>
                        ))
                    : <p className={`${styles.tr} ${styles.massages}`}>Нямата запазен час за масаж</p>
            }
            {
                massages.length > 5
                    ?
                    massages.length - 5 > 1
                        ? <h4 className={`${styles.tr} ${styles.massages} ${styles.more}`}>
                            Имате още {massages.length - 5} предстоящи масажи!</h4>
                        : <h4 className={`${styles.tr} ${styles.massages} ${styles.more}`}>
                            Имате още {massages.length - 5} предстоящ масаж!</h4>
                    : null
            }
        </div>
    )
}

export default UserMassages
