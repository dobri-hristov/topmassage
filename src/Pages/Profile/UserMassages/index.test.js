import React from 'react'
import TestRenderer from 'react-test-renderer'
import UserMassages from './index'

describe('Profile page/UserMassages component', () => {
    test('should render correctly with NO data', () => {
        const component = TestRenderer.create(<UserMassages massages={[]}/>)
        const tree = component.toJSON()
        expect(tree).toMatchSnapshot()
    })

    test('should render correctly with data (length < 5)', () => {
        const data = [
            {
                massage: 'Класически',
                date: '25-11-2021',
                hour: '11:00'
            }
        ]
        const component = TestRenderer.create(<UserMassages massages={data} />)
        const tree = component.toJSON()
        expect(tree).toMatchSnapshot()
    })

    test('should render correctly with data (length = 6)', () => {
        const data = [
            {
                massage: 'Класически',
                date: '25-11-2021',
                hour: '11:00'
            },
            {
                massage: 'Масаж на глава',
                date: '25-11-2021',
                hour: '12:00'
            },
            {
                massage: 'Класически',
                date: '25-11-2021',
                hour: '13:00'
            },
            {
                massage: 'Масаж на глава',
                date: '25-11-2021',
                hour: '14:00'
            },
            {
                massage: 'Класически',
                date: '25-11-2021',
                hour: '15:00'
            },
            {
                massage: 'Масаж на глава',
                date: '25-11-2021',
                hour: '16:00'
            }
        ]
        const component = TestRenderer.create(<UserMassages massages={data}/>)
        const tree = component.toJSON()
        expect(tree).toMatchSnapshot()
    })

    test('should render correctly with data (length > 6)', () => {
        const data = [
            {
                massage: 'Класически',
                date: '25-11-2021',
                hour: '11:00'
            },
            {
                massage: 'Масаж на глава',
                date: '25-11-2021',
                hour: '12:00'
            },
            {
                massage: 'Класически',
                date: '25-11-2021',
                hour: '13:00'
            },
            {
                massage: 'Масаж на глава',
                date: '25-11-2021',
                hour: '14:00'
            },
            {
                massage: 'Класически',
                date: '25-11-2021',
                hour: '15:00'
            },
            {
                massage: 'Масаж на глава',
                date: '25-11-2021',
                hour: '16:00'
            },
            {
                massage: 'Масаж на глава',
                date: '25-11-2021',
                hour: '17:00'
            }
        ]
        const component = TestRenderer.create(<UserMassages massages={data}/>)
        const tree = component.toJSON()
        expect(tree).toMatchSnapshot()
    })
})