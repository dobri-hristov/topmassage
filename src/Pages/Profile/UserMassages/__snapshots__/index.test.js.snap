// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile page/UserMassages component should render correctly with NO data 1`] = `
<div
  className="massages-form"
>
  <h3
    className="title"
  >
    Предстоящи масажи
  </h3>
  <div
    className="tr trMain"
  >
    <h3
      className="type"
    >
      Вид масаж
    </h3>
    <h3>
      Дата
    </h3>
    <h3
      className="hour"
    >
      Час
    </h3>
  </div>
  <p
    className="tr massages"
  >
    Нямата запазен час за масаж
  </p>
</div>
`;

exports[`Profile page/UserMassages component should render correctly with data (length < 5) 1`] = `
<div
  className="massages-form"
>
  <h3
    className="title"
  >
    Предстоящи масажи
  </h3>
  <div
    className="tr trMain"
  >
    <h3
      className="type"
    >
      Вид масаж
    </h3>
    <h3>
      Дата
    </h3>
    <h3
      className="hour"
    >
      Час
    </h3>
  </div>
  <p
    className="tr massages"
  >
    Нямата запазен час за масаж
  </p>
</div>
`;

exports[`Profile page/UserMassages component should render correctly with data (length = 6) 1`] = `
<div
  className="massages-form"
>
  <h3
    className="title"
  >
    Предстоящи масажи
  </h3>
  <div
    className="tr trMain"
  >
    <h3
      className="type"
    >
      Вид масаж
    </h3>
    <h3>
      Дата
    </h3>
    <h3
      className="hour"
    >
      Час
    </h3>
  </div>
  <p
    className="tr massages"
  >
    Нямата запазен час за масаж
  </p>
</div>
`;

exports[`Profile page/UserMassages component should render correctly with data (length > 6) 1`] = `
<div
  className="massages-form"
>
  <h3
    className="title"
  >
    Предстоящи масажи
  </h3>
  <div
    className="tr trMain"
  >
    <h3
      className="type"
    >
      Вид масаж
    </h3>
    <h3>
      Дата
    </h3>
    <h3
      className="hour"
    >
      Час
    </h3>
  </div>
  <p
    className="tr massages"
  >
    Нямата запазен час за масаж
  </p>
</div>
`;
